src/snowflake/snowpark/session.py

```diff
@@ -351,6 +351,19 @@ def _remove_config(self, key: str) -> "Session.SessionBuilder":
         def app_name(self, app_name: str) -> "Session.SessionBuilder":
             """
             Adds the app name to the :class:`SessionBuilder` to set in the query_tag after session creation
+
+            Note:
+                Since version ``1.19.1``, the app name is set to the query tag in JSON format. For example:
+
+                >>> session = Session.builder.app_name("my_app").configs(db_parameters).create() # doctest: +SKIP
+                >>> print(session.query_tag) # doctest: +SKIP
+                {"APPNAME": "my_app"}
+
+                In previous versions it is set using a key=value format. For example:
```

Comment:
we usually don't keep track of version differences in the doc string. `CHANGELOG.md` is the right place for that.

Reply:
Hi @sfc-gh-aalam, Could you take a look at this Pull Request again, please?